src/Symfony/Component/Validator/Constraints/CollectionValidator.php

```diff
@@ -34,7 +35,7 @@ public function validate($value, Constraint $constraint)
         }
 
         if (!\is_array($value) && !($value instanceof \Traversable && $value instanceof \ArrayAccess)) {
-            throw new UnexpectedTypeException($value, 'array or Traversable and ArrayAccess');
+            throw new UnexpectedValueException($value, 'array or Traversable and ArrayAccess');
```

Comment:
i still think we need to tackle this being translator friendly, it'll only work for english

Reply:
e.g. different messages like

- `This value should be of type {{ type }}` (generic)
- `This value should be array accessible`
- `This value should be iterable`
- `This value should be a countable`

Reply:
what about using `array|(iterable&ArrayAccess)`?

Reply:
sounds like a plan yes :) the messages dont have to be pretty (tailored), but should be correct.

Reply:
We don't translate exception messages

Reply:
the issue happens here: https://github.com/symfony/symfony/pull/27917/files#diff-d46c902789b437519780ee8110ecb5e9R807

Reply:
@ostrolucky we do, see https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Violation/ConstraintViolationBuilder.php#L149